src/main/mw.c

```diff
@@ -723,159 +617,222 @@ void filterGyro(void) {
+#ifndef USE_QUAD_MIXER_ONLY
+            && !((masterConfig.mixerMode == MIXER_TRI || masterConfig.mixerMode == MIXER_CUSTOM_TRI) && masterConfig.mixerConfig.tri_unarmed_servo)
+            && masterConfig.mixerMode != MIXER_AIRPLANE
+            && masterConfig.mixerMode != MIXER_FLYING_WING
+#endif
+    ) {
+        rcCommand[YAW] = 0;
+    }
+
+    if (currentProfile->throttle_correction_value && (FLIGHT_MODE(ANGLE_MODE) || FLIGHT_MODE(HORIZON_MODE))) {
+        rcCommand[THROTTLE] += calculateThrottleAngleCorrection(currentProfile->throttle_correction_value);
+    }
 
-        // if GPS feature is enabled, gpsThread() will be called at some intervals to check for stuck
-        // hardware, wrong baud rates, init GPS if needed, etc. Don't use SENSOR_GPS here as gpsThread() can and will
-        // change this based on available hardware
 #ifdef GPS
-        if (feature(FEATURE_GPS)) {
-            gpsThread();
+    if (sensors(SENSOR_GPS)) {
+        if ((FLIGHT_MODE(GPS_HOME_MODE) || FLIGHT_MODE(GPS_HOLD_MODE)) && STATE(GPS_FIX_HOME)) {
+            updateGpsStateForHomeAndHoldMode();
         }
-#endif
     }
+#endif
 
-    currentTime = micros();
-    if (masterConfig.looptime == 0 || (int32_t)(currentTime - loopTime) >= 0) {
-        loopTime = currentTime + masterConfig.looptime;
+    // PID - note this is function pointer set by setPIDController()
+    pid_controller(
+        &currentProfile->pidProfile,
+        currentControlRateProfile,
+        masterConfig.max_angle_inclination,
+        &currentProfile->accelerometerTrims,
+        &masterConfig.rxConfig
+    );
 
-        imuUpdate(&currentProfile->accelerometerTrims);
+    mixTable();
 
-        // Measure loop rate just after reading the sensors
-        currentTime = micros();
-        cycleTime = (int32_t)(currentTime - previousTime);
-        previousTime = currentTime;
+#ifdef USE_SERVOS
+    filterServos();
+    writeServos();
+#endif
 
-        dT = (float)cycleTime * 0.000001f;
+    if (motorControlEnable) {
+        writeMotors();
+    }
 
-        if (currentProfile->pidProfile.gyro_cut_hz) {
-            filterGyro();
-        }
+#ifdef BLACKBOX
+    if (!cliMode && feature(FEATURE_BLACKBOX)) {
+        handleBlackbox();
+    }
+#endif
+}
 
-        annexCode();
+void taskUpdateAccelerometer(uint32_t deltaTime)
+{
+    UNUSED(deltaTime);
+    imuUpdateAccelerometer(&currentProfile->accelerometerTrims);
+}
 
-        if (masterConfig.rxConfig.rcSmoothing) {
-            filterRc();
-        }
+void taskHandleSerial(uint32_t deltaTime)
+{
+    UNUSED(deltaTime);
+    handleSerial();
+}
 
-#if defined(BARO) || defined(SONAR)
-        haveProcessedAnnexCodeOnce = true;
-#endif
+void taskUpdateBeeper(uint32_t deltaTime)
+{
+    UNUSED(deltaTime);
+    beeperUpdate();          //call periodic beeper handler
+}
 
-#ifdef MAG
-        if (sensors(SENSOR_MAG)) {
-            updateMagHold();
-        }
-#endif
+void taskUpdateBattery(uint32_t deltaTime)
```

Comment:
with new code, it is possible to plan updateBattery at VBATINTERVAL and updateCurrentMeter at IBATINTERVAL .. but that can wait